ssh integration tests fail in the Quarkus Platform

### Bug description

This code in `SshTestResource` does not work:

https://github.com/apache/camel-quarkus/blob/1618ae47eb936fa1f7dbaab30a28719f3e86ef43/integration-tests/ssh/src/test/java/org/apache/camel/quarkus/component/ssh/it/SshTestResource.java#L82-L83

```
[ERROR] org.apache.camel.quarkus.component.ssh.it.SshTest.testWriteToSSHAndReadFromSSH  Time elapsed: 0.002 s  <<< ERROR!
java.lang.RuntimeException: java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at io.quarkus.test.junit.QuarkusTestExtension.throwBootFailureException(QuarkusTestExtension.java:611)
	at io.quarkus.test.junit.QuarkusTestExtension.interceptTestClassConstructor(QuarkusTestExtension.java:695)
	at java.base/java.util.Optional.orElseGet(Optional.java:364)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
Caused by: java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at io.quarkus.test.junit.TestResourceUtil$TestResourceManagerReflections.startReflectively(TestResourceUtil.java:214)
	at io.quarkus.test.junit.QuarkusTestExtension.doJavaStart(QuarkusTestExtension.java:212)
	at io.quarkus.test.junit.QuarkusTestExtension.ensureStarted(QuarkusTestExtension.java:578)
	at io.quarkus.test.junit.QuarkusTestExtension.beforeAll(QuarkusTestExtension.java:628)
	... 1 more
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at io.quarkus.test.junit.TestResourceUtil$TestResourceManagerReflections.startReflectively(TestResourceUtil.java:211)
	... 4 more
Caused by: java.util.concurrent.CompletionException: java.lang.RuntimeException: Unable to start Quarkus test resource class org.apache.camel.quarkus.component.ssh.it.SshTestResource
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:315)
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:320)
	at java.base/java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1807)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: java.lang.RuntimeException: Unable to start Quarkus test resource class org.apache.camel.quarkus.component.ssh.it.SshTestResource
	at io.quarkus.test.common.TestResourceManager$TestResourceRunnable.run(TestResourceManager.java:674)
	at java.base/java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1804)
	... 3 more
Caused by: java.lang.RuntimeException: java.nio.file.FileSystemNotFoundException
	at org.apache.camel.quarkus.component.ssh.it.SshTestResource.start(SshTestResource.java:102)
	at io.quarkus.test.common.TestResourceManager$TestResourceRunnable.run(TestResourceManager.java:668)
	... 4 more
Caused by: java.nio.file.FileSystemNotFoundException
	at jdk.zipfs/jdk.nio.zipfs.ZipFileSystemProvider.getFileSystem(ZipFileSystemProvider.java:156)
	at jdk.zipfs/jdk.nio.zipfs.ZipFileSystemProvider.getPath(ZipFileSystemProvider.java:142)
	at java.base/java.nio.file.Path.of(Path.java:209)
	at java.base/java.nio.file.Paths.get(Paths.java:98)
	at org.apache.camel.quarkus.component.ssh.it.SshTestResource.start(SshTestResource.java:83)
	... 5 more
```

--- a/integration-tests/ssh/src/test/java/org/apache/camel/quarkus/component/ssh/it/SshTestResource.java
+++ b/integration-tests/ssh/src/test/java/org/apache/camel/quarkus/component/ssh/it/SshTestResource.java
@@ -16,6 +16,10 @@
  */
 package org.apache.camel.quarkus.component.ssh.it;
 
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
 import java.nio.file.Paths;
 import java.util.LinkedList;
 import java.util.List;
@@ -79,8 +83,16 @@ public class SshTestResource implements QuarkusTestResourceLifecycleManager {
 
             sshd = SshServer.setUpDefaultServer();
             sshd.setPort(edPort);
-            sshd.setKeyPairProvider(new FileKeyPairProvider(
-                    Paths.get(Thread.currentThread().getContextClassLoader().getResource("edDSA/key_ed25519.pem").toURI())));
+
+            File tmpFile = File.createTempFile("key_ed25519-", ".pem");
+            try (InputStream in = Thread.currentThread().getContextClassLoader().getResourceAsStream("edDSA/key_ed25519.pem")) {
+                Files.write(tmpFile.toPath(), in.readAllBytes());
+                sshd.setKeyPairProvider(new FileKeyPairProvider(tmpFile.toPath()));
+            } catch (NullPointerException | IOException e) {
+                String message = "An issue occurred while loading key: edDSA/key_ed25519.pem";
+                throw new RuntimeException(message, e);
+            }
+
             sshd.setCommandFactory(new TestEchoCommandFactory());
             sshd.setPasswordAuthenticator((username, password, session) -> true);
             sshd.setPublickeyAuthenticator((username, key, session) -> true);

